### DISCUSSION TITLE
should it be stated that `php_sodium.dll` is also required since `lcobucci/jwt` requires it?

### DISCUSSION THREAD

**Discussion by rebbieboi** [kreait/firebase-php#954] [General]
as the question states should it be stated in the doc that php_sodium.dll is required because of lcobucci/jwt? currently it only says mbstring as a requirement

**Comment by jeromegamez** [comment]
Platform-specific requirements of another package are unrelated to the SDK, so I shouldn't add it to the SDK's dependencies. As far as I know, the requirements of all packages are merged, so if I don't have a platform-specific requirement, it should still be one because of the other library.

Besides, I wouldn't add DLLs directly as requirements. For example, if I require the `intl` PHP extension, you need ICU as a system dependency, but I wouldn't (and couldn't) add it as requirement on the `composer.json` file.

I hope this makes sense.
